src/resources/ble_esp32.cc

```diff
@@ -647,12 +666,28 @@ class BleRemoteDeviceResource : public ServiceContainer<BleRemoteDeviceResource>
   uint16 handle() const { return handle_; }
   void set_handle(uint16 handle) { handle_ = handle; }
 
+  int connect(uint8 own_addr_type, ble_addr_t* addr) {
+    int err = ble_gap_connect(own_addr_type, addr, 3000, null,
```

Comment:
What should happen if we're already connected?

Reply:
I added a check that returns `BLE_ERR_SUCCESS` if it is already connected.